File: src/kyu_6/what_century_is_it/python/what_century_is_it.py
from math import ceil

SUFFIX = {1: 'st', 2: 'nd', 3: 'rd'}

def what_century(year):
    century = int(ceil(int(year) / 100.0))
    quo, rem = divmod(century, 10)
    if quo == 1 and 1 <= rem <= 3:
        suffix = 'th'
    else:
        suffix = SUFFIX.get(rem, 'th')
    return '{}{}'.format(century, suffix)

File: src/kyu_6/what_century_is_it/csharp/what_century_is_it.cs
public class Kata
{
  public static string WhatCentury(string year)
  {
        int century = int.Parse(year);
        if (century % 100 == 0) {
            century = century / 100;
        } else {
            century = (century / 100) + 1;
        }
        string postfix = "th";
        if (century % 10 == 1) {
            postfix = "st";
        }
        if (century % 10 == 2) {
            postfix = "nd";
        }
        if (century % 10 == 3) {
            postfix = "rd";
        }
        if (century > 10 && century <= 13) {
            postfix = "th";
        }

        return century + postfix;
  }
}